cassovary-core/src/main/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraph.scala

```diff
@@ -115,7 +118,8 @@ case class BipartiteSide(nodes: Array[BipartiteNode], numOfNodes: Int, numOfOutE
 class BipartiteGraph(val leftNodes: Array[BipartiteNode], val leftNodeCount: Int,
                      val leftOutEdgeCount: Long, val rightNodes: Array[BipartiteNode],
                      val rightNodeCount: Int, val rightOutEdgeCount: Long,
-                     val bipartiteGraphDir: BipartiteGraphDir.BipartiteGraphDir) extends Graph[BipartiteNode] {
+                     val bipartiteGraphDir: BipartiteGraphDir.BipartiteGraphDir,
+                     val isLeftNode: (Int => Option[Boolean]) = (_ => None)) extends Graph[BipartiteNode] {
```

Comment:
This also means that is unique is false, new leftNodes and rightNodes will need to be constructed with the correct ids. I don't see a way around that.